CDCgov/dibbs-query-connector
# Save condition selections on template builder page

# PULL REQUEST

## Summary
- Update condition selection page to:
  - Autofocus the Query Name input on page load/refresh (if the input is blank)
  - Indicate that Query Name is required
- Add `selectedConditions` to component's state
  - Toggling the checkbox will now add/remove the given condition from this group 

## Related Issue

Fixes #118 

## Additional Information
- Original AC was to save data from this page to the db; since we are/were not saving query-condition relations, this isn't appropriate. Instead, the data is saved in the component state and will be used to retrieve the corresponding value sets on the next page (eventually sent to the db with the full query data (value sets, concepts, etc)).

## How to Test
- Navigate to the "My Queries" page: `landing page` -> `gear icon` -> `sign in` -> `gear icon` -> `my queries` -> `create query button`
  - The "Query Name" input should autofocus and the "Create Query" button should be disabled
  -  On hover over the button, tooltip text should read "Select at least one condition below"
- Select one or more conditions from at least two categories
  - The "Create Query" button should now be enabled
- Click to create the query, without adding a query name. 
  - An inline error should appear and focus should jump back to the Query Name input 
  - Tooltip text should read "Enter a query name, then click to create your query"
- Name your query
  - The inline error should go away
  - Tooltip text should read "Click to create your query"
- Open the dev console
- Click "Create Query" and confirm all your selections are logged appropriately
- Uncheck one condition and repeat above
  - Confirm the unchecked item was removed
  - Repeat such that a given category no longer has any checked items; confirm the category is no longer logged in the console 

Ex:
<img width="400" alt="Screenshot 2024-11-14 at 17 19 46" src="https://github.com/user-attachments/assets/6b3ca290-c970-4629-ace3-3751c1f0e7c1">
<img width="400" alt="Screenshot 2024-11-14 at 17 20 03" src="https://github.com/user-attachments/assets/1aafc455-2fe9-4243-879c-d0255cd5ce3f">

## Checklist

- [x] Descriptive Pull Request title
- [x] Link to relevant issues
- [x] Provide necessary context for design reviewers
